libs/partners/cohere/langchain_cohere/chat_models.py

```diff
@@ -181,16 +189,29 @@ async def _astream(
                 if run_manager:
                     await run_manager.on_llm_new_token(delta, chunk=chunk)
                 yield chunk
+            elif data.event_type == "stream-end":
+                generation_info = self._get_generation_info(data.response)
+                yield ChatGenerationChunk(
+                    message=AIMessageChunk(
+                        content="", additional_kwargs=generation_info
+                    ),
+                    generation_info=generation_info,
+                )
 
     def _get_generation_info(self, response: Any) -> Dict[str, Any]:
         """Get the generation info from cohere API response."""
-        return {
+        generation_info = {
             "documents": response.documents,
             "citations": response.citations,
             "search_results": response.search_results,
             "search_queries": response.search_queries,
-            "token_count": response.token_count,
+            "is_search_required": response.is_search_required,
+            "generation_id": response.generation_id,
+            "tool_calls": response.tool_calls,
```

Comment:
let's factor this into the same format as other models
```
tool_calls = [
  {
    "id": "some id string",
    "function": {
      "name": ...,
      "arguments": "json serialized dict or args {}"
    }
  }
]
```

Reply:
Without doing this, this will also mess with serialization

Reply:
This is implemented, thank you!